src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -159,6 +159,34 @@ private CentralPackageVersion ToCentralPackageVersion(string packageId, string v
             return new CentralPackageVersion(packageId, VersionRange.Parse(version));
         }
 
+        private IReadOnlyDictionary<string, PrunePackageReference> GetPackagesToPrune()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            IEnumerable<(string PackageId, string Version)> packageVersions =
+                        _vsProjectAdapter.GetBuildItemInformation(ProjectItems.PrunePackageReference, ProjectBuildProperties.Version)
+                        .Select(item => (PackageId: item.ItemId, Version: item.ItemMetadata.FirstOrDefault()));
+
+            return packageVersions
+                .Select(item => ToPrunePackageReference(item.PackageId, item.Version))
+                .ToDictionary(cpv => cpv.Name, StringComparer.OrdinalIgnoreCase);
+
+            static PrunePackageReference ToPrunePackageReference(string id, string versionString)
```

Comment:
There are a few different places where the constructor to `PrunePackageReference` is called with a string and `VersionRange` via `NuGetVersion.Parse()` which sets `includeMaxVersion`.  I think it would be better to just have a constructor to `PrunePackageReference` or static method that takes two strings and handles all of the validation.

```cs
// All xml docs here calling out exceptions
public PrunePackageReference(string id, string version)
{
    if (string.IsNullOrEmpty(id))
    {
      throw new ArgumentNullException(nameof(id));
    }

    if (string.IsNullOrEmpty(versionString))
    {
      throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.Error_PrunePackageReferenceMissingVersion, id));
    }
  
    Name = name;
    VersionRange = new VersionRange(maxVersion: NuGetVersion.Parse(version), includeMaxVersion: true);
}
```

Then all of the places that need the PrunePackageReference could just call that constructor, and this static method isn't needed.

https://github.com/NuGet/NuGet.Client/pull/6159/files#diff-9d32a7ae6b417fc1b23cef544c58cea9aac005c8539bdee7d430c4004a9ab7a4R617
https://github.com/NuGet/NuGet.Client/pull/6159/files#diff-a1e0e948fb5d1439e0368357d8587e83fafe5d5bb1a4f3fad40b040b3b9bf022R392
https://github.com/NuGet/NuGet.Client/pull/6159/files#diff-69bcc4cc18dd020ef093e419c357cd2a3452094ec27be239a03c665b32de863bR788

Reply:
That's a good idea. I'll look into and re-request review when ready.